cmd/boskos/boskos.go

```diff
@@ -68,12 +66,14 @@ var (
 	configPath = flag.String("config", "config.yaml", "Path to init resource file")
 	_          = flag.Duration("dynamic-resource-update-period", defaultDynamicResourceUpdatePeriod,
 		"Legacy flag that does nothing but is kept for compatibility reasons")
-	requestTTL              = flag.Duration("request-ttl", defaultRequestTTL, "request TTL before losing priority in the queue")
-	logLevel                = flag.String("log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
-	namespace               = flag.String("namespace", corev1.NamespaceDefault, "namespace to install on")
-	port                    = flag.Int("port", 8080, "Port to serve on")
-	enableLeaderElection    = flag.Bool("enable-leader-election", false, "If leader election should be enabled. If enabled, it is possible to run Boskos with multiple replicas and only the one that is leader will receive traffic.")
-	leaderElectionNamespace = flag.String("leader-election-namespace", "", "Namespace to use for leader election. If running in-cluster, this defaults to the pods namespace")
+	requestTTL           = flag.Duration("request-ttl", defaultRequestTTL, "request TTL before losing priority in the queue")
+	logLevel             = flag.String("log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+	namespace            = flag.String("namespace", corev1.NamespaceDefault, "namespace to install on")
+	port                 = flag.Int("port", 8080, "Port to serve on")
+	enableLeaderElection = flag.Bool("enable-leader-election", false, "If leader election should be enabled. If enabled, it is possible to run Boskos with multiple replicas and only the one that is leader will receive traffic. This works through boskos managing a label so it is only set on the leader pod. The boskos service must select on this label and not one that matches all replicas.")
+	boskosLabelSelector  = flag.String("boskos-label-selector", "", "Must be set when leader election is used. The label selector to identify boskos pods")
+	podName              = flag.String("pod-name", "", "The name of the current pod. Required when --enable-leadere-election is set.")
```

Comment:
This is a breaking change for existing boskos instances using leader election. We should probably avoid this by retaining the old behavior unless opted in by specifying these flags.

Reply:
The leader election mechanism was only added very recently in https://github.com/kubernetes-sigs/boskos/pull/99, didn't solve the problem it was supposed to solve and this only breaks ppl if they opted in  - I would argue that this is unlikely to break anyone.